### ISSUE TITLE
Sensor pins not working on RPI after power failure reboot

### ISSUE THREAD

**Issue by ToddGreenfield** [KraigM/homebridge-gpio-garagedoor#14]
I have a weird problem here. When my RPI reboots (only from a power failure, but not when restart command given), the switches/relay/door openers work fine, but the sensors that detect state do not show a change in state, they always show closed- unless... I run the below python repo script once and then view status on that scripts webpage. Once that script is started and webpage viewed, I then stop the python script and start homebridge. This plugin loads and everything then works perfect, door status and switches. 

Originally I was playing around with https://github.com/andrewshilliday/garage-door-controller, which is a python script to operate my garage doors and why this is involved-  but I later opted to use this plugin (hb plugin). The python script works both with sensors and switches.

Suggestions?

**Comment by troystanway** [comment]
Did you end up working this out? I had some issues with this before, but I found the GPIO that I used was the issue. I run a reed switch from 3v3 pin to a GPIO, when the reed is together the 3v flows to the  GPIO.

**Comment by ToddGreenfield** [comment]
Not yet. I haven't messed with it for a while, but saw someone else suggested to use different GPIOs as well. Maybe I will play with it for a bit this afternoon.

**Comment by ToddGreenfield** [comment]
So this just proves one needs to take a few minutes to read/learn a bit more about things they are not 100% confident with. I was able to fix this by changing the sensor pins to GPIOs 2 & 3 which are pulled high at boot time (at least for my RpiZ). The pins I was previously using were 17 & 27 which are pulled low at boot, which is apparently why the state never changed until I ran the afore mentioned python script which pulled them high. A little research on which pins are set high/low at boot did the trick. Thanks for the hint Troy.
